### PR TITLE
Fix "Zoom in" and "View these X" drilldown actions for charts based on custom columns

### LINKED ISSUE #13289: Cannot use "Zoom in" action drill-through when grouped by Custom Column
**Describe the bug**
When a question is grouped by a Custom Column, then it's not possible to use "Zoom in" drill-through action.

**To Reproduce**
1. Custom question > Sample Dataset > Orders
2. Custom Column `1 + 1` as "TestColumn"
3. Summarize Count by "TestColumn" and CreatedAt:Month
image
4. Visualize and click any of the line data points and click "Zoom in" - fails with
`Value does not match schema: {:query {:breakout [(named [nil (named (named (not (some-matching-condition? a-clojure.lang.PersistentVector)) "Must be a valid instance of one of these clauses: :field-id, :field-literal, :fk->, :joined-field") "field") nil] "Must be a valid instance of one of these clauses: :field-id, :field-literal, :joined-field, :fk->, :datetime-field, :expression, :binning-strategy")]}}`

<details><summary>Full stacktrace</summary>

```
09-22 11:16:39 ERROR middleware.catch-exceptions :: Error processing query: null
{:database_id 4,
 :started_at #t "2020-09-22T11:16:39.715090+02:00[Europe/Copenhagen]",
 :error_type :invalid-query,
 :json_query
 {:type "query",
  :query
  {:source-table 11,
   :expressions {:TestColumn ["+" 1 1]},
   :aggregation [["count"]],
   :filter ["and" ["=" ["expression" "TestColumn"] 2] ["=" ["datetime-field" ["field-id" 86] "month"] "2018-01-01T00:00:00+01:00"]],
   :breakout [["datetime-field" ["expression" "TestColumn"] "week"]]},
  :database 4,
  :parameters [],
  :middleware {:js-int-to-string? true, :add-default-userland-constraints? true}},
 :native nil,
 :status :failed,
 :class clojure.lang.ExceptionInfo,
 :stacktrace
 ["--> util.schema$schema_core_validator$fn__21086.invoke(schema.clj:26)"
  "query_processor.middleware.validate$validate_query$fn__47230.invoke(validate.clj:9)"
  "query_processor.middleware.normalize_query$normalize$fn__45648.invoke(normalize_query.clj:22)"
  "query_processor.middleware.add_rows_truncated$add_rows_truncated$fn__39280.invoke(add_rows_truncated.clj:36)"
  "query_processor.middleware.results_metadata$record_and_return_metadata_BANG_$fn__47197.invoke(results_metadata.clj:147)"
  "query_processor.middleware.constraints$add_default_userland_constraints$fn__44927.invoke(constraints.clj:42)"
  "query_processor.middleware.process_userland_query$process_userland_query$fn__46389.invoke(process_userland_query.clj:136)"
  "query_processor.middleware.catch_exceptions$catch_exceptions$fn__44870.invoke(catch_exceptions.clj:174)"
  "query_processor.reducible$async_qp$qp_STAR___38074$thunk__38075.invoke(reducible.clj:101)"
  "query_processor.reducible$async_qp$qp_STAR___38074.invoke(reducible.clj:107)"
  "query_processor.reducible$sync_qp$qp_STAR___38083$fn__38086.invoke(reducible.clj:133)"
  "query_processor.reducible$sync_qp$qp_STAR___38083.invoke(reducible.clj:132)"
  "query_processor$process_userland_query.invokeStatic(query_processor.clj:215)"
  "query_processor$process_userland_query.doInvoke(query_processor.clj:211)"
  "query_processor$fn__47372$process_query_and_save_execution_BANG___47381$fn__47384.invoke(query_processor.clj:227)"
  "query_processor$fn__47372$process_query_and_save_execution_BANG___47381.invoke(query_processor.clj:219)"
  "query_processor$fn__47416$process_query_and_save_with_max_results_constraints_BANG___47425$fn__47428.invoke(query_processor.clj:239)"
  "query_processor$fn__47416$process_query_and_save_with_max_results_constraints_BANG___47425.invoke(query_processor.clj:232)"
  "api.dataset$fn__50707$fn__50710.invoke(dataset.clj:55)"
  "query_processor.streaming$streaming_response_STAR_$fn__35496$fn__35497.invoke(streaming.clj:73)"
  "query_processor.streaming$streaming_response_STAR_$fn__35496.invoke(streaming.clj:72)"
  "async.streaming_response$do_f_STAR_.invokeStatic(streaming_response.clj:66)"
  "async.streaming_response$do_f_STAR_.invoke(streaming_response.clj:64)"
  "async.streaming_response$do_f_async$fn__23282.invoke(streaming_response.clj:85)"],
 :context :ad-hoc,
 :error
 "Value does not match schema: {:query {:breakout [(named [nil (named (named (not (some-matching-condition? a-clojure.lang.PersistentVector)) \"Must be a valid instance of one of these clauses: :field-id, :field-literal, :fk->, :joined-field\") \"field\") nil] \"Must be a valid instance of one of these clauses: :field-id, :field-literal, :joined-field, :fk->, :datetime-field, :expression, :binning-strategy\")]}}",
 :row_count 0,
 :running_time 0,
 :preprocessed nil,
 :ex-data
 {:type :schema.core/error,
  :value
  {:type :query,
   :query
   {:source-table 11,
    :expressions {:TestColumn [:+ 1 1]},
    :aggregation [[:count]],
    :filter [:and [:= [:expression "TestColumn"] 2] [:= [:datetime-field [:field-id 86] :month] "2018-01-01T00:00:00+01:00"]],
    :breakout [[:datetime-field [:expression "TestColumn"] :week]]},
   :database 4,
   :middleware {:js-int-to-string? true, :add-default-userland-constraints? true},
   :info
   {:executed-by 1,
    :context :ad-hoc,
    :nested? false,
    :query-hash [20, -75, 21, 70, -78, 70, -112, -113, 54, -25, -77, 37, -120, -52, 1, -43, 98, 61, -124, 99, -22, -88, 9, 36, -48, -78, -18, 34, -98, 81, -113, 34]},
   :constraints {:max-results 10000, :max-results-bare-rows 2000}},
  :error
  {:query
   {:breakout
    [(named [nil (named (named (not (some-matching-condition? a-clojure.lang.PersistentVector)) "Must be a valid instance of one of these clauses: :field-id, :field-literal, :fk->, :joined-field") "field") nil] "Must be a valid instance of one of these clauses: :field-id, :field-literal, :joined-field, :fk->, :datetime-field, :expression, :binning-strategy")]}}},
 :data {:rows [], :cols []}}

09-22 11:16:39 DEBUG middleware.log :: POST /api/dataset 202 [ASYNC: completed] 118.0 ms (3 DB calls) App DB connections: 0/10 Jetty threads: 3/50 (3 idle, 0 queued) (82 total active threads) Queries in flight: 0 (0 queued)
```

</details>

**Information about your Metabase Installation:**
Metabase 0.34.2, 0.35.3 and 0.36.6

**Additional context**
The error is partially similar to #12938, but I don't think they are related much.

### PR DESCRIPTION
**Pull request by daltojohnso**
Fixes #13289 

Fixes the "View these X" and "Zoom" drilldown actions. As part of this I am hiding two "Automatic exploration" actions that should show, because they are still broken (see bug https://github.com/metabase/metabase/issues/16680)

Here's a recording of how to create the proper line chart and proof of fix:

https://user-images.githubusercontent.com/13057258/122608129-91ea6f80-d030-11eb-88ee-29e91a28760a.mov

### CODE REVIEW

**Comment by daltojohnso** [comment]
Lemme write some tests for this

**Comment by daltojohnso** [comment]
I'm thinking that the "View these X" filters that the drilldown chart receive shouldn't include the "two is 2" filter--it's more obviously wrong when the custom column values differ.

**Comment by daltojohnso** [comment]
Need to look into something so making this a WIP

File: frontend/src/metabase/modes/components/drill/AutomaticDashboardDrill.jsx [hunk 1]
```diff
@@ -15,10 +16,17 @@ export default ({ question, clicked }: ClickActionProps): ClickAction[] => {
...
+  const includesExpressionDimensions = dimensions.some(dimension => {
+    return isExpressionField(dimension.column.field_ref);
+  });
+
   if (
     !clicked ||
     dimensions.length === 0 ||
-    !MetabaseSettings.get("enable-xrays")
+    !MetabaseSettings.get("enable-xrays") ||
+    includesExpressionDimensions
```
Comment (gusaiani) [inline frontend/src/metabase/modes/components/drill/AutomaticDashboardDrill.jsx:29]: Do these chained conditions mean something that can be named?

Might be nice to have them as a constant, with a declarative name.

File: frontend/src/metabase/modes/components/drill/AutomaticDashboardDrill.jsx [hunk 2]
```diff
@@ -15,10 +16,17 @@ export default ({ question, clicked }: ClickActionProps): ClickAction[] => {
 
   // questions with a breakout
   const dimensions = (clicked && clicked.dimensions) || [];
+
+  // ExpressionDimensions don't work right now (see metabase#16680)
```
Comment (gusaiani) [inline frontend/src/metabase/modes/components/drill/AutomaticDashboardDrill.jsx:20]: This function, which is repeated below, looks like a great candidate to live in a shared `util`-type file.

**Review by gusaiani — APPROVED** [review]
Working nicely now 👍

A couple of inlined comments.

I'd encourage a follow-up PR adding propTypes to touched files missing them.

File: frontend/src/metabase/modes/components/drill/AutomaticDashboardDrill.jsx (hunk 1 above)
Comment (daltojohnso) [reply frontend/src/metabase/modes/components/drill/AutomaticDashboardDrill.jsx]: done

File: frontend/src/metabase/modes/components/drill/AutomaticDashboardDrill.jsx (hunk 2 above)
Comment (daltojohnso) [reply frontend/src/metabase/modes/components/drill/AutomaticDashboardDrill.jsx:20]: this code will get deleted as part of #16680 so I think duplicated logic is fine here

**Review by gusaiani — APPROVED** [review]
🚀

**Pull request merged**